File: javascript/series/series.js
export class Series {
  constructor(string) {
    this.digitsArray = string.split("").map(element => Number(element));
  }

  get digits() {
    return this.digitsArray;
  }

  slices(n) {
    if (this.digitsArray.length < n) {
      throw new Error("Slice size is too big.");
    }

    let seriesArray = [];

    for (let i = 0; i <= this.digitsArray.length - n; i++) {
      seriesArray.push(this.digitsArray.slice(i, i + n));
    }
    return seriesArray;
  }
}

File: ruby/series/series.rb
class Series
    def initialize(number)
        @number = number
    end

    def slices(n)
        if @number.length < n
            raise ArgumentError.new
        else
            slicesArray = []
            max = @number.length-n

            for i in(0..max)
                slicesArray.push(@number.slice(i..i+n-1))
            end

            slicesArray
        end      
    end
end

